hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCancelAirdropHandler.java

```diff
@@ -50,14 +51,18 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {}
 
     @Override
-    public void handle(@NonNull final HandleContext context) throws HandleException {
-        var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
-        validateTrue(tokensConfig.cancelTokenAirdropEnabled(), NOT_SUPPORTED);
-    }
+    public void handle(@NonNull final HandleContext context) throws HandleException {}
 
     @NonNull
     @Override
     public Fees calculateFees(@NonNull final FeeContext feeContext) {
-        return Fees.FREE;
+        var tokensConfig = feeContext.configuration().getConfigData(TokensConfig.class);
+        validateTrue(tokensConfig.cancelTokenAirdropEnabled(), NOT_SUPPORTED);
```

Comment:
We should throw this in Ingest

Reply:
Had a conversation with Michael H. about this: https://limechain.slack.com/archives/C079T1M2KQF/p1719987291918269. Do you agree with that or do we need to discuss it further?